api/requester/requester.go

```diff
@@ -75,139 +80,201 @@ func SetVerbose(verb bool) {
 	verbose = verb
 }
 
-// PostParams represents params struct
-type PostParams = map[string]interface{}
+// PlatformRequest represents params struct
+type PlatformRequest struct {
+	JSONRPC        string      `json:"jsonrpc"`
+	ID             int         `json:"id"`
+	Method         string      `json:"method"`
+	PlatformParams interface{} `json:"params"`
+	LogLevel       string      `json:"logLevel,omitempty"`
+}
 
-// GetResponseBody makes request and extracts body
-func GetResponseBody(url string, postP PostParams) ([]byte, error) {
+// GetResponseBodyContract makes request to contract and extracts body
+func GetResponseBodyContract(url string, postP Request, signature string) ([]byte, error) {
 	jsonValue, err := json.Marshal(postP)
 	if err != nil {
-		return nil, errors.Wrap(err, "[ getResponseBody ] Problem with marshaling params")
+		return nil, errors.Wrap(err, "[ getResponseBodyContract ] Problem with marshaling params")
 	}
 
 	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonValue))
 	if err != nil {
-		return nil, errors.Wrap(err, "[ getResponseBody ] Problem with creating request")
+		return nil, errors.Wrap(err, "[ getResponseBodyContract ] Problem with creating request")
 	}
 	req.Header.Set("Content-Type", "application/json")
+
+	h := sha256.New()
+	_, err = h.Write(jsonValue)
+	if err != nil {
+		return nil, errors.Wrap(err, "[ getResponseBodyContract ] Cant get hash")
+	}
+	sha := base64.URLEncoding.EncodeToString(h.Sum(nil))
+	req.Header.Set("Digest", "SHA-256="+sha)
```

Comment:
You use this headers more then in one place. Move to constants.
I mean "Digest", "Signature" ...
Might be something else